### ISSUE TITLE
segfault in systemd-analyze with specific service file

### ISSUE THREAD

**Issue by oddlama** [systemd/systemd#23389]
**systemd version the issue has been seen with**
> systemd 250 (250)
> +PAM -AUDIT -SELINUX -APPARMOR +IMA +SMACK +SECCOMP +GCRYPT -GNUTLS +OPENSSL +ACL +BLKID -CURL -ELFUTILS -FIDO2 +IDN2 -IDN -IPTC +KMOD -LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY -P11KIT -QRENCODE -BZIP2 +LZ4 -XZ -ZLIB +ZSTD -BPF_FRAMEWORK -XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified

**Used distribution**
 > gentoo, fully updated on 2022-05-14

**Linux kernel version used** (`uname -a`)

 > `Linux 5.17.7-gentoo-dist #1 SMP PREEMPT Thu May 12 12:38:00 -00 2022 x86_64 AMD EPYC Processor (with IBPB) AuthenticAMD GNU/Linux`

**CPU architecture issue was seen on**
 > amd64 (AMD Ryzen Threadripper 1950X 16-Core Processor)

**Expected behaviour you didn't see**
 > No segfault.

**Unexpected behaviour you saw**
 > Segfault when using `systemd-analyze security ntfy.service`

**Steps to reproduce the problem**

Copy the provided `ntfy.service` file to `/lib/systemd/system/ntfy.service` and execute `systemd-analyze security ntfy.service`.

```ntfy.service```:

```
[Unit]
Description=ntfy server
Documentation=https://github.com/binwiederhier/ntfy
After=network.target

[Service]
User=ntfy
Group=ntfy
ExecStart=/usr/bin/ntfy serve
Restart=on-failure

LimitNOFILE=10000
LimitNPROC=256

PrivateTmp=true
PrivateDevices=true
ProtectHome=true
ProtectSystem=strict
ProtectKernelTunables=yes
ProtectKernelModules=yes
ProtectControlGroups=yes

RestrictNamespaces=yes

SystemCallArchitectures=native
SystemCallFilter=@system-service
RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6

WorkingDirectory=/var/lib/ntfy
ReadWriteDirectories=/var/lib/ntfy
ReadWriteDirectories=/var/cache/ntfy
# Allow ntfy to bind ports in the range of 0-1024
AmbientCapabilities=CAP_NET_BIND_SERVICE
CapabilityBoundingSet=CAP_NET_BIND_SERVICE

[Install]
WantedBy=multi-user.target
```

**Additional program output to the terminal or log subsystem illustrating the issue**

```
Bus n/a: changing state UNSET → OPENING
sd-bus: starting bus by connecting to /run/systemd/private...
Bus n/a: changing state OPENING → AUTHENTICATING
Successfully forked off '(pager)' as PID 3162195.
Found cgroup2 on /sys/fs/cgroup/, full unified hierarchy
Failed to execute 'pager', using next fallback pager: No such file or directory
Pager executable is "less", options "FRSXMK", quit_on_interrupt: yes
Bus n/a: changing state AUTHENTICATING → RUNNING
Sent message type=method_call sender=n/a destination=org.freedesktop.systemd1 path=/org/freedesktop/systemd1/unit/ntfy_2eservice interface=org.freedesktop.DBus.Properties member=GetAll cookie=1 reply_cookie=0 signature=s error-name=n/a err>
Got message type=method_return sender=org.freedesktop.systemd1 destination=n/a path=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 signature=a{sv} error-name=n/a error-message=n/a
[1]    3162194 segmentation fault (core dumped)  SYSTEMD_LOG_LEVEL=debug systemd-analyze security ntfy.service
```

I was able to reproduce this on another system. If you cannot reproduce this, I can provide you with the relevant coredump file.

**Comment by yuwata** [comment]
This should be already fixed by 444d9abd06478525fc5da3b03c512f6a1f0ef12b.

**Comment by bluca** [comment]
yes, same issue
